test/jdk/javax/swing/JTabbedPane/bug4703690.java

```diff
@@ -44,105 +45,103 @@
-            InvocationTargetException {
+    public static void main(String[] args) throws Exception {
         bug4703690 test = new bug4703690();
         try {
             SwingUtilities.invokeAndWait(() -> {
-                fr = new JFrame("bug4703690");
+                frame = new JFrame("bug4703690");
 
-                panel = new JPanel();
+                JPanel panel = new JPanel();
                 one = new JButton("Button 1");
                 panel.add(one);
                 two = new JButton("Button 2");
                 panel.add(two);
 
                 tabbedPane = new JTabbedPane();
-                fr.getContentPane().add(tabbedPane);
+                frame.getContentPane().add(tabbedPane);
                 tabbedPane.addTab("Tab one", panel);
                 tabbedPane.addTab("Tab two", new JPanel());
 
                 two.addFocusListener(new FocusAdapter() {
                     public void focusGained(FocusEvent e) {
-                        focusButtonTwo = true;
+                        focusButtonTwo.countDown();
                     }
                 });
 
                 tabbedPane.addChangeListener(e -> {
                     if (tabbedPane.getSelectedIndex() == 1) {
-                        switchToTabTwo = true;
+                        switchToTabTwo.countDown();
                     }
                 });
 
-                fr.setBounds(10, 10, 200, 200);
-                fr.setVisible(true);
-                fr.setLocationRelativeTo(null);
+                frame.setSize(200, 200);
+                frame.setLocationRelativeTo(null);
+                frame.setVisible(true);
             });
 
             test.execute();
         } finally {
             SwingUtilities.invokeAndWait(() -> {
-                if (fr != null) {
-                    fr.dispose();
+                if (frame != null) {
+                    frame.dispose();
                 }
             });
         }
     }
 
-    public void execute() {
-        try {
-            robot = new Robot();
-            robot.setAutoDelay(50);
-            robot.delay(1000);
-            two.requestFocus();
-
-            one.addFocusListener(new FocusAdapter() {
-                    public void focusGained(FocusEvent e) {
-                        focusButtonOne = true;
-                    }
-                });
-
-            SwingUtilities.invokeAndWait(() -> {
-                p = tabbedPane.getLocationOnScreen();
-                rect = tabbedPane.getBoundsAt(1);
-            });
-
-            robot.delay(1000);
-            robot.mouseMove(p.x + rect.x + rect.width / 2,
-                            p.y + rect.y + rect.height / 2);
-            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
-            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+    public void execute() throws Exception {
+        robot = new Robot();
+        robot.setAutoDelay(50);
 
-            robot.delay(1000);
+        two.requestFocus();
```

Comment:
If you say that EDT is better, this call should also be called on EDT.

Reply:
Added this line to EDT. Thanks again for the help with revamping this test!